.github/workflows/push_tests.yml

```
@@ -0,0 +1,45 @@
+name: Run all tests
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  HF_HOME: /mnt/cache
+  OMP_NUM_THREADS: 8
+  MKL_NUM_THREADS: 8
+  PYTEST_TIMEOUT: 180
```

Think we can increase this here (we're not nearly at the size of Transformers)
```suggestion
  PYTEST_TIMEOUT: 1000
```